packages/jest-haste-map/src/crawlers/node.ts

```diff
@@ -193,10 +202,8 @@ export = function nodeCrawl(
       });
     };
 
-    if (forceNodeFilesystemAPI || process.platform === 'win32') {
-      find(roots, extensions, ignore, callback);
-    } else {
-      findNative(roots, extensions, ignore, callback);
-    }
+    hasNativeFindSupport(forceNodeFilesystemAPI)
+      .then(() => findNative(roots, extensions, ignore, callback))
+      .catch(() => find(roots, extensions, ignore, callback));
```

Comment:
this needs an extra `catch` that rejects the promise we return, in case it fails in some other way while crawling